source/switch/main.cpp

```diff
@@ -623,6 +624,13 @@ int main(int argc, char *argv[]) {
 				uiPopState();
 				if (uiGetState() == stateRunning) uiPopState();
 				break;
+			case resultOpenSettings:
+				if (uiGetState() != stateRunning) {
+					uiPopState();
+					uiPushState(stateSettings);
```

Comment:
this seems odd to me. Why do you pop the pause state and later push it again when you could also just push the settings state and later pop it?